### DISCUSSION TITLE
Messagebox popup icon does not match the ico icon of the parent window? How can I set it to be the same?

### DISCUSSION THREAD

**Discussion by zhenzi0322** [israel-dryer/ttkbootstrap#348] [Q&A] [answered]
如下图所示：
![image](https://user-images.githubusercontent.com/34839719/189259581-a08d007e-f08a-4b9f-bc16-fcb58b4c6d10.png)

示例如下：
```python
import ttkbootstrap as ttk
from tkinter import messagebox
from ttkbootstrap.constants import *
from ttkbootstrap.dialogs.dialogs import Messagebox

root = ttk.Window()
root.iconbitmap("favicon.ico")
root.wm_minsize(1280, 700)


def open_ttkbootstrap_warning():
    Messagebox.show_warning("Hello")

b1 = ttk.Button(root, text="ttkbootstrap show_warning", bootstyle=SUCCESS, command=open_ttkbootstrap_warning)
b1.pack(padx=5, pady=10)


root.mainloop()
```

**Comment by Megsman** [comment]
try `root.iconbitmap(default = 'favicon.ico')`

**Comment by israel-dryer** [comment, marked as answer]
This is the same root cause as #1095 (recently fixed/closed): `ttk.Window` sets ttkbootstrap's icon as the *application-wide default* (`iconphoto(True, ...)`), so the Messagebox popup inherits that instead of the `.ico` you set on the parent window.

@Megsman's suggestion is on the right track — `iconbitmap(default=...)` is the Windows mechanism that propagates an icon to child dialogs. Two ways to fix it:

```python
# Option A — disable ttkbootstrap's default icon, then set your own
root = ttk.Window(iconphoto=None)
root.iconbitmap("favicon.ico")

# Option B — set your .ico as the app-wide default (Windows)
root.iconbitmap(default="favicon.ico")
```

Both make the popup's titlebar icon match the parent.

Heads-up: **2.0** reworks icon handling so this mostly "just works" — pass your icon to the constructor and it's applied via `wm_iconbitmap(default=...)`, which child dialogs (Messagebox, pickers) inherit automatically:

```python
root = ttk.Window(iconphoto="favicon.ico")
```
